sample/redo/src/login.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {useRef, useEffect, useCallback} from 'react';
+
+import {loginURL} from './settings';
+
+export type LoginResult = {
+  email: string;
+  userId: string;
+};
+
+async function loggedInUser(): Promise<LoginResult | undefined> {
```

Comment:
[cbb?] Canyou say `?LoginResult` or something, like what works for params and fields?

Reply:
TS does not have a shorthand for or null or undefined.